tests/tensorflow/test_tensorflow_autolog.py

```diff
@@ -0,0 +1,213 @@
+# pep8: disable=E501
+
+from __future__ import print_function
+
+import collections
+import shutil
+import pytest
+import tempfile
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.python.keras import layers
+
+import mlflow
+import mlflow.tensorflow
+
+SavedModelInfo = collections.namedtuple(
+        "SavedModelInfo",
+        ["path", "meta_graph_tags", "signature_def_key", "inference_df", "expected_results_df"])
+
+
+@pytest.fixture
+def tf_keras_random_data():
+    mlflow.tensorflow.autolog(metrics_every_n_steps=2)
+    def random_one_hot_labels(shape):
+        n, n_class = shape
+        classes = np.random.randint(0, n_class, n)
+        labels = np.zeros((n, n_class))
+        labels[np.arange(n), classes] = 1
+        return labels
+
+    data = np.random.random((1000, 32))
+    labels = random_one_hot_labels((1000, 10))
+
+    model = tf.keras.Sequential()
+
+    model.add(layers.Dense(64, activation='relu'))
+    model.add(layers.Dense(64, activation='relu'))
+    model.add(layers.Dense(10, activation='softmax'))
+
+    model.compile(optimizer=tf.train.AdamOptimizer(0.001),
+                  loss='categorical_crossentropy',
+                  metrics=['accuracy'])
+    run = mlflow.start_run()
+    model.fit(data, labels, epochs=10)
+    mlflow.end_run()
+    return run.info.run_id
+
+
+@pytest.fixture
+def get_tf_keras_random_data_run_instance(tf_keras_random_data):
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(tf_keras_random_data)
+    return run
+
+
+@pytest.mark.small
```

Comment:
Tests are marked as small by default, so we can get rid of these annotations

Reply:
Also a nit, maybe we can just combine all of these tests that assert on metrics/params/tags into a single test, like `test_tf_keras_logs_expected_data` - it's just a bit less verbose & makes it easier to see in one place what we expect to have logged etc.